src/main/java/com/rultor/agents/IndexesRequests.java

```diff
@@ -39,8 +47,106 @@
  * @version $Id$
  */
 public final class IndexesRequests implements SuperAgent {
+    /**
+     * Xpath expression for retrieving log entries from a talk's XML.
+     */
+    private static final String ARCHIVE_LOG = "//archive/log";
+
+    /**
+     * Name of the log index attribute.
+     */
+    private static final String INDEX = "index";
+
     @Override
+    @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
     public void execute(final Talks talks) throws IOException {
-        throw new UnsupportedOperationException();
+        int max = this.max(talks);
+        for (final Talk talk : talks.active()) {
+            max = this.update(talk, max);
+        }
+    }
+
+    /**
+     * Creates an index node in the specified talk node, if it has archive/log
+     * nodes.
+     * @param talk Talk, to which the index node should be appended.
+     * @param max Maximal index value among all talks
+     * @return New maximal value. If an index node was added, it is equal to
+     *  (max+1), otherwise - max.
+     * @throws IOException Thrown in case of XML parsing errors.
+     */
+    private int update(final Talk talk, final int max) throws IOException {
+        final List<String> requests = talk.read().xpath("//request");
+        final List<XML> logs = talk.read().nodes(ARCHIVE_LOG);
+        int newmax = max;
+        int index = this.max(talk);
+        if (index < 1) {
+            index = max;
+        }
+        if (requests.isEmpty() && !logs.isEmpty()) {
+            talk.modify(
+                new Directives().xpath("//talk").add("request")
+                    .attr(INDEX, Integer.toString(index + 1))
+                    .attr("id", this.createRequestId())
+                    .add("type").set(INDEX)
+                    .up()
+                    .add("args")
+            );
+            newmax += 1;
+        }
+        return newmax;
+    }
+
+    /**
+     * Calculates maximum index value of all archive/log nodes.
+     * @param talk Talk, for whose archve/log nodes the maximum should be
+     *  calculated.
+     * @return Maximum value of index attribute among all archive/log nodes of
+     *  talk.
+     * @throws IOException Thrown in case of XML parsing problems.
+     */
+    private int max(final Talk talk) throws IOException {
+        final Iterable<Integer> indexes = Iterables.transform(
+            talk.read()
+                .xpath("/talk/archive/log/@index|/talk/request/@index"),
+            new Function<String, Integer>() {
+                @Override
+                public Integer apply(final String input) {
+                    return Integer.parseInt(input);
+                }
+            });
```

Comment:
closing `)` bracket should be at the next line, see http://www.yegor256.com/2014/10/23/paired-brackets-notation.html